e2e/ansible/playbooks/resiliency/test-k8s-master-failure/k8s-master-failure.yml

```diff
@@ -0,0 +1,315 @@
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: jenkins_password
+
+
+     - name: 5a) Restart the jenkins pod for running jenkins actions.
+       shell: >
+          source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+          -- java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 restart
+          --username admin --password {{ jenkins_password.stdout }}
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+ 
+     - name: wait for 60 seconds.
+       wait_for:
+          timeout: 60
+
+
+     - name: 6)  Create and Build one sample job (Hello World)  inside pod.
+       shell: >
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }} 
+         -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s
+         http://localhost:8080 create-job sample < /var/jenkins_home/template.xml";
+         source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace  }} 
+         -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 build sample"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+
+
+     - name: 6a)  Running a build job. 
+       shell: >
+          source ~/.profile; kubectl exec {{ jenkins_pod_name.stdout }} -n {{ namespace }}
+          -- bash -c "java -jar /var/jenkins_home/jenkins-cli.jar -s http://localhost:8080 console sample"
+       args:
+         executable: /bin/bash
+       delegate_to: "{{groups['kubernetes-kubemasters'].0}}"
+       register: First_Build
+
+
+     - name: Bring down the kubemaster.
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/vagrant-halt-up-task.yml"
+       vars:
+         vagrant_op: halt --force
+         nodename: kubemaster-01
+
+
+     - name: wait for 120 seconds.
+       wait_for:
+          timeout: 120
+
+
+
+     - name: 7) Checking the status of kubemaster after kubemaseter halt.
+       shell: vagrant status kubemaster-01 | grep kubemaster-01 |  awk '{ $1="";print}' | cut -d '(' -f1
+       args:
+         chdir: "{{ansible_env.HOME}}/{{vagrant_path}}"
+         executable: /bin/bash
+       register: VDown
+       until: "'poweroff'  in VDown.stdout"
+
+
+     - name: Bring up the kubemaster.
+       include_tasks: "{{ansible_env.HOME}}/{{utils_path}}/vagrant-halt-up-task.yml"
+       vars:
+         vagrant_op: up 
+         nodename: kubemaster-01
+
+     - name: wait for 120 seconds.
+       wait_for:
+          timeout: 120
+
+     - name: 7a) Checking the status of kubemaster.
+       shell: vagrant status kubemaster-01 | grep kubemaster-01 |  awk '{ $1="";print}' | cut -d '(' -f1
+       args:
+         chdir: "{{ansible_env.HOME}}/{{vagrant_path}}"
+         executable: /bin/bash
+       register: Vup
+       until: "' running '  in Vup.stdout"
+
+
```

Comment:
Can we push all the vagrant related tasks to the a separate task file - as seen in the node failure testcase ? We should have the test generic and executable on cloud as well. Having separate task files will help us conditionally include it based on the nature of the test bed

Reply:
Yes, we can do that, but we are including test into run-test.yml ...to skip any test which is having vagrant operations.